thoth/python/python_artifact.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# thoth-python
+# Copyright(C) 2019 Kevin Postlthwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Representation a python module and all the files within."""
+
+import shutil
+import logging
+import requests
+import tempfile
+import zipfile
+import tarfile
+import hashlib
+import os
+from elftools.elf.elffile import ELFFile
+from elftools.common.exceptions import ELFError
+from typing import Iterator, Tuple
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PythonArtifact:
+    """Python artifacts are compressed modules."""
+
+    def __init__(self, artifact_name, artifact_url, compressed_file_name=None, verify_ssl=False):
+        """Create a new Python Artifact."""
+        self.verify_ssl = verify_ssl
+        self.artifact_name = artifact_name
+        self.artifact_url = artifact_url
+        self.dir_name = None
+
+        if compressed_file_name is not None:
+            self.compressed_file = open(compressed_file_name, "r+b")
```

Comment:
In most cases, its not a good idea to keep an opened file as an attribute.

Reply:
I wasn't sure how to go about this because when a tempfile is closed it is deleted.  I could create my own random naming schema and save to `/tmp/<randomly_generated>` if you think that would be better

Reply:
Use `NamedTemporaryFile` and set `delete` to False. It will be then your responsibility to remove the file once not needed [1].

[1] https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

Reply:
Thanks, didn't know there was a `delete` optional variable